### PR TITLE
feat: add keyboard navigation to calendar view (#924)

### LINKED ISSUE #924: feat: add keyboard navigation to calendar view
## Description

The calendar view is the only database view without keyboard navigation. Board, gallery, list, and table views all have dedicated keyboard hooks (`board-keyboard.ts`, `gallery-keyboard.ts`, `list-keyboard.ts`, `table-keyboard.ts`) and E2E keyboard specs. Calendar view has ARIA roles (`role="grid"`, `role="gridcell"`) but no `onKeyDown` handler — users cannot navigate between days or items using the keyboard.

## Acceptance Criteria

- [ ] Arrow keys navigate between calendar day cells (Left/Right move between days, Up/Down move between weeks)
- [ ] Home/End move to the first/last day of the current week
- [ ] Enter on a focused day cell with items focuses the first item; Enter on a focused item navigates to the row page
- [ ] Escape returns focus from an item back to the day cell
- [ ] Tab moves focus out of the calendar grid (standard tab order)
- [ ] Focused cell has a visible focus ring matching the design spec (`--accent` ring)
- [ ] Navigation wraps at month boundaries (arrow past last day focuses first day of next month, and vice versa)
- [ ] Keyboard hook extracted to `src/components/database/views/calendar-keyboard.ts` following the pattern of `list-keyboard.ts` and `gallery-keyboard.ts`
- [ ] Unit tests for the keyboard hook in `calendar-keyboard.test.ts`
- [ ] E2E spec `e2e/database-calendar-keyboard.spec.ts` covering arrow navigation, Enter to open item, Escape to return
- [ ] `pnpm lint && pnpm typecheck && pnpm test` pass

## Dependencies

None

## Technical Notes

- Follow the pattern in `src/components/database/views/gallery-keyboard.ts` — it uses a grid-based navigation model similar to what calendar needs
- The calendar grid already uses `role="grid"` and `role="gridcell"` ARIA attributes (line 299, 365 of `calendar-view.tsx`)
- The container has `tabIndex={-1}` and a `ref` (line 238) — wire the `onKeyDown` handler there
- Calendar cells are rendered via `buildCalendarGrid()` from `calendar-view-helpers.ts` — use the grid structure for row/column index tracking
- Each cell has a `date` string and optional items — the keyboard hook needs to track `focusedDate` and optionally `focusedItemIndex`
- Reference `.agents/conventions.md` for component patterns

### PR DESCRIPTION
**Pull request by sw-factory-automations**
Closes #924

## What

Adds full keyboard navigation to the calendar view, making it the last database view to gain keyboard support. The calendar grid already had ARIA roles (`role="grid"`, `role="gridcell"`) but no `onKeyDown` handler.

## How

- **New hook:** `useCalendarKeyboardNavigation` in `calendar-keyboard.ts`, following the pattern of `gallery-keyboard.ts` and `list-keyboard.ts`. Tracks a two-level focus state: `cellIndex` (which day cell) and `itemIndex` (which item within the cell, or `null` for cell-level focus).
- **Cell navigation:** Arrow Left/Right move between adjacent cells. Arrow Up/Down jump by 7 (one week). Home/End move to the first/last day of the current week row. Navigation past grid boundaries calls `onPrevMonth`/`onNextMonth`.
- **Item navigation:** Enter on a cell with items focuses the first item. Arrow Up/Down moves between items within a cell. Enter on a focused item navigates to the row page via `onNavigate`. Escape returns focus from item to cell.
- **Focus ring:** Focused cells and items show `ring-2 ring-ring ring-offset-0` (accent color), matching the gallery view pattern.
- **Integration:** Wired `onNavigate={router.push}` into the CalendarView usage in `database-view-client.tsx`.
- **Bug fix:** Prevented focus event bubbling from child items to the cell's `onFocus` handler (which would reset `itemIndex` to `null`).

## Testing

- 25 unit tests in `calendar-keyboard.test.ts` covering all navigation keys, item-level navigation, boundary wrapping, and edge cases.
- 7 E2E tests in `e2e/database-calendar-keyboard.spec.ts` covering arrow navigation, Home/End, Enter to focus items, Enter to navigate to page, Escape to return, and focus ring visibility.
- `pnpm lint && pnpm typecheck && pnpm test` — all pass (131 files, 1791 tests).
- Existing `e2e/database-calendar.spec.ts` — all 4 tests still pass.

### CODE REVIEW

File: src/components/database/views/calendar-view.tsx [hunk 1]
```diff
@@ -364,12 +369,24 @@ function CalendarDayCell({ cell, workspaceSlug, onClick }: CalendarDayCellProps)
...
       className={cn(
-        "relative min-h-24 border border-overlay-border p-1",
+        "relative min-h-24 border border-overlay-border p-1 outline-none",
         cell.isToday && "bg-accent/10",
         onClick && "cursor-pointer",
+        isFocused &&
+          focusedItemIndex === null &&
+          "ring-2 ring-ring ring-offset-0",
       )}
       onClick={onClick ? () => onClick(cell.date) : undefined}
```
Comment (sw-factory-automations) [inline src/components/database/views/calendar-view.tsx:383]: The focus ring uses `ring-offset-0` but the design spec (`.agents/design.md` → Accessibility) specifies `ring-2 ring-ring ring-offset-2`, and all other database views (gallery, list, board) use `ring-offset-2`.

Additionally, the gallery/list/board views include `focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2` as a CSS class alongside the programmatic `isFocused` ring. This ensures a focus ring appears even for native browser focus (e.g., tabbing). The calendar cell is missing this.

Suggested fix:
```tsx
className={cn(
  "relative min-h-24 border border-overlay-border p-1 outline-none",
  "focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  cell.isToday && "bg-accent/10",
  onClick && "cursor-pointer",
  isFocused &&
    focusedItemIndex === null &&
    "ring-2 ring-ring ring-offset-2",
)}
```

File: src/components/database/views/calendar-view.tsx [hunk 2]
```diff
@@ -514,17 +533,29 @@ function CalendarMobileList({ days, workspaceSlug, onClick }: CalendarMobileList
...
 
   return (
     <Link
       href={`/${workspaceSlug}/${row.page.id}`}
       onClick={(e) => e.stopPropagation()}
-      className="mb-0.5 block truncate bg-muted px-1 py-0.5 text-xs hover:bg-overlay-active"
+      tabIndex={isFocused ? 0 : -1}
+      data-calendar-item={itemIndex}
+      className={cn(
+        "mb-0.5 block truncate bg-muted px-1 py-0.5 text-xs outline-none hover:bg-overlay-active",
```
Comment (sw-factory-automations) [inline src/components/database/views/calendar-view.tsx:557]: Same issue here — `ring-offset-0` should be `ring-offset-2` to match the design spec and other views. Also missing `focus:ring-*` classes for native focus.

Suggested fix:
```tsx
className={cn(
  "mb-0.5 block truncate bg-muted px-1 py-0.5 text-xs outline-none hover:bg-overlay-active",
  "focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  isFocused && "ring-2 ring-ring ring-offset-2",
)}
```

**Comment by sw-factory-automations** [comment]
**Review — 3 issues to fix before merge:**

1. **Focus ring offset mismatch** — Both the cell and item focus rings use `ring-offset-0`, but the design spec (`.agents/design.md` → Accessibility) specifies `ring-offset-2`, and all other database views (gallery, list, board) use `ring-offset-2`. See inline comments for suggested fixes.

2. **Missing native focus ring CSS** — Gallery, list, and board views include `focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2` as CSS classes on focusable elements, ensuring a ring appears for native browser focus (e.g., Tab key). The calendar cells and items only show a ring via the programmatic `isFocused` prop. Add the `focus:ring-*` classes to match the other views.

3. **Stories not updated** — `calendar-view.stories.tsx` should include `onNavigate: fn()` in the default args, matching the gallery and list view stories. This enables keyboard navigation testing in Storybook.

Fixing these now.

**Comment by sw-factory-automations** [comment]
✅ UI verification passed — design spec compliance confirmed.

**Static analysis** — all checks pass:

| Check | Result |
|---|---|
| Color tokens | ✅ No arbitrary hex/rgb. Uses `bg-accent/10`, `bg-overlay-active`, `ring-ring`, `bg-muted`, `border-overlay-border`. |
| Typography | ✅ `text-xs` for items, `text-sm` for day numbers, `text-lg` only on `<h2>` heading. |
| Spacing | ✅ All Tailwind scale values (`p-1`, `mb-0.5`, `px-1`, `py-0.5`, `gap-2`). No arbitrary spacing. |
| Button variants | ✅ `ghost` for nav arrows, `default` for Today button. |
| Focus rings | ✅ `ring-2 ring-ring ring-offset-2` matches design spec (line 592). |
| Accessibility | ✅ `aria-label` on cells and grid, `role="gridcell"`, `role="grid"`, proper `tabIndex` management, focus bubbling prevention. |
| Dark mode | ✅ CSS variable tokens throughout. |
| Borders | ✅ `border-overlay-border`. Shadow only on overflow popover (floating element exception). |
| Corners | ✅ Sharp corners on cells/items. `rounded-sm` only on overflow popover (floating element exception). |
| Transitions | ✅ No decorative animations added. |
| Loading states | ✅ Skeleton placeholders (not spinners). |

**Storybook visual regression** — calendar view stories pass. Unrelated failures:
- 8 missing baselines for `ViewConfigDropdown` stories (from a prior PR, not this one).
- 1 time-dependent diff on `date-renderer--editor-no-date` (current month changed since baseline capture).

**Storybook visual review** — 12 calendar stories screenshotted (Default, Empty, Filtered Empty, Loading, No Date Property, With Overflow, Many Items, Read Only, Mobile, Mobile Overflow, Tablet). All render correctly with proper spacing, token usage, and responsive behavior.

**Comment by sw-factory-automations** [comment]
✅ Post-merge verification passed.

**E2E tests (live site):** 19/19 passed
- `e2e/database-calendar-keyboard.spec.ts` — 7 tests (ArrowRight/Left, ArrowUp/Down, Home/End, Enter to focus items, Enter to navigate, Escape to return, focus ring visibility)
- `e2e/database-calendar.spec.ts` — 4 tests (view switching, item placement, month navigation, row creation)
- `e2e/auth.spec.ts` — 8 tests (sign-in, sign-up, sign-out, redirects, callbacks)

**Ad-hoc smoke tests:** All passed
- ✓ Landing page loads with title
- ✓ Sign-in page renders with email input
- ✓ Health endpoint returns OK
- ✓ Authenticated login flow succeeds
- ✓ No console errors (unauthenticated or authenticated)
- Skipped: `/dashboard` (not yet built), page editor navigation (no pages in test workspace)

**Interaction smoke test:** Calendar keyboard navigation — skipped (test workspace has no database pages to interact with). Covered by E2E tests above which create their own test data.
